### ISSUE TITLE
Build instructions for macOS and Linux need updating

### ISSUE THREAD

**Issue by faultylee** [symless/synergy#5529]
Hi, I found that compiling v1.8.2 on my Arch (server) shows some unusual error at the end of `conf` and `build` and gui wasn't built. Might be missing some files

When I run `bin/synergys`, synergy seems to connect and work, thought I've no GUI to manage it. 

Error snippet from `QT_SELECT=4 ./hm.sh conf -g1`
```
Going back to: /mnt/data/build/synergy-v1.8.2-stable-36cd521-Source
Error: Could not get revision, git error: 128
```

Error snippet from `./hm.sh build`
```
Make GUI command: make -w
Entering dir: src/gui
make: Entering directory '/mnt/data/build/synergy-v1.8.2-stable-36cd521-Source/src/gui'
make: *** No targets specified and no makefile found.  Stop.
make: Leaving directory '/mnt/data/build/synergy-v1.8.2-stable-36cd521-Source/src/gui'
Going back to: /mnt/data/build/synergy-v1.8.2-stable-36cd521-Source
Error: make -w failed with error: 512
```

Full `QT_SELECT=4 ./hm.sh conf -g1` output
```
Mapping command: conf -> configure
Running setup...
Setup complete.
cmake version 3.6.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
Creating dir: build/release
Entering dir: build/release
CMake command: cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release ../..
-- The C compiler identification is GNU 6.1.1
-- The CXX compiler identification is GNU 6.1.1
-- Check for working C compiler: /usr/lib/hardening-wrapper/bin/cc
-- Check for working C compiler: /usr/lib/hardening-wrapper/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/lib/hardening-wrapper/bin/c++
-- Check for working CXX compiler: /usr/lib/hardening-wrapper/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include istream
-- Looking for C++ include istream - found
-- Looking for C++ include ostream
-- Looking for C++ include ostream - found
-- Looking for C++ include sstream
-- Looking for C++ include sstream - found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file locale.h
-- Looking for include file locale.h - found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file sys/select.h
-- Looking for include file sys/select.h - found
-- Looking for include file sys/socket.h
-- Looking for include file sys/socket.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file sys/utsname.h
-- Looking for include file sys/utsname.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file wchar.h
-- Looking for include file wchar.h - found
-- Looking for getpwuid_r
-- Looking for getpwuid_r - found
-- Looking for gmtime_r
-- Looking for gmtime_r - found
-- Looking for nanosleep
-- Looking for nanosleep - found
-- Looking for poll
-- Looking for poll - found
-- Looking for sigwait
-- Looking for sigwait - found
-- Looking for strftime
-- Looking for strftime - found
-- Looking for vsnprintf
-- Looking for vsnprintf - found
-- Looking for inet_aton
-- Looking for inet_aton - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of char
-- Check size of char - done
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of short
-- Check size of short - done
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found CURL: /usr/lib64/libcurl.so (found version "7.50.1") 
-- Looking for XRRNotifyEvent
-- Looking for XRRNotifyEvent - not found
-- Looking for 3 include files X11/Xlib.h, ..., X11/extensions/dpms.h
-- Looking for 3 include files X11/Xlib.h, ..., X11/extensions/dpms.h - found
-- Looking for include file X11/extensions/Xinerama.h
-- Looking for include file X11/extensions/Xinerama.h - found
-- Looking for 3 include files X11/Xlib.h, ..., X11/extensions/XKBstr.h
-- Looking for 3 include files X11/Xlib.h, ..., X11/extensions/XKBstr.h - found
-- Looking for include file X11/extensions/XKB.h
-- Looking for include file X11/extensions/XKB.h - found
-- Looking for include file X11/extensions/XTest.h
-- Looking for include file X11/extensions/XTest.h - found
-- Looking for include files X11/Xlib.h, X11/XKBlib.h
-- Looking for include files X11/Xlib.h, X11/XKBlib.h - found
-- Looking for include file X11/extensions/XInput2.h
-- Looking for include file X11/extensions/XInput2.h - found
-- Looking for IceConnectionNumber in SM;ICE
-- Looking for IceConnectionNumber in SM;ICE - found
-- Looking for DPMSQueryExtension in Xext;X11
-- Looking for DPMSQueryExtension in Xext;X11 - found
-- Looking for XTestQueryExtension in Xtst;Xext;X11
-- Looking for XTestQueryExtension in Xtst;Xext;X11 - found
-- Looking for XineramaQueryExtension in Xinerama
-- Looking for XineramaQueryExtension in Xinerama - found
-- Looking for XISelectEvents in Xi
-- Looking for XISelectEvents in Xi - found
-- Looking for XRRQueryExtension in Xrandr
-- Looking for XRRQueryExtension in Xrandr - found
-- Configuring done
-- Generating done
-- Build files have been written to: /mnt/data/build/synergy-v1.8.2-stable-36cd521-Source/build/release
Going back to: /mnt/data/build/synergy-v1.8.2-stable-36cd521-Source
Error: Could not get revision, git error: 128
```
Full `./hm.sh build` output
```
Entering dir: build/release
Scanning dependencies of target arch
[  0%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/Arch.cpp.o
[  1%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/ArchConsoleStd.cpp.o
[  1%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/ArchDaemonNone.cpp.o
[  2%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/IArchString.cpp.o
[  2%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchConsoleUnix.cpp.o
[  3%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchDaemonUnix.cpp.o
[  3%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchFileUnix.cpp.o
[  4%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchInternetUnix.cpp.o
[  5%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchLogUnix.cpp.o
[  5%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchMultithreadPosix.cpp.o
[  6%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchNetworkBSD.cpp.o
[  6%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchPluginUnix.cpp.o
[  7%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchSleepUnix.cpp.o
[  7%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchStringUnix.cpp.o
[  8%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchSystemUnix.cpp.o
[  8%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchTaskBarXWindows.cpp.o
[  9%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/ArchTimeUnix.cpp.o
[ 10%] Building CXX object src/lib/arch/CMakeFiles/arch.dir/unix/XArchUnix.cpp.o
[ 10%] Linking CXX static library ../../../../../lib/libarch.a
[ 10%] Built target arch
Scanning dependencies of target common
[ 11%] Building CXX object src/lib/common/CMakeFiles/common.dir/PluginVersion.cpp.o
[ 11%] Building CXX object src/lib/common/CMakeFiles/common.dir/Version.cpp.o
[ 12%] Linking CXX static library ../../../../../lib/libcommon.a
[ 12%] Built target common
Scanning dependencies of target base
[ 13%] Building CXX object src/lib/base/CMakeFiles/base.dir/Event.cpp.o
[ 13%] Building CXX object src/lib/base/CMakeFiles/base.dir/EventQueue.cpp.o
[ 14%] Building CXX object src/lib/base/CMakeFiles/base.dir/EventTypes.cpp.o
[ 14%] Building CXX object src/lib/base/CMakeFiles/base.dir/FunctionEventJob.cpp.o
[ 15%] Building CXX object src/lib/base/CMakeFiles/base.dir/FunctionJob.cpp.o
[ 16%] Building CXX object src/lib/base/CMakeFiles/base.dir/Log.cpp.o
[ 16%] Building CXX object src/lib/base/CMakeFiles/base.dir/SimpleEventQueueBuffer.cpp.o
[ 17%] Building CXX object src/lib/base/CMakeFiles/base.dir/Stopwatch.cpp.o
[ 17%] Building CXX object src/lib/base/CMakeFiles/base.dir/String.cpp.o
[ 18%] Building CXX object src/lib/base/CMakeFiles/base.dir/Unicode.cpp.o
[ 18%] Building CXX object src/lib/base/CMakeFiles/base.dir/XBase.cpp.o
[ 19%] Building CXX object src/lib/base/CMakeFiles/base.dir/log_outputters.cpp.o
[ 19%] Linking CXX static library ../../../../../lib/libbase.a
[ 19%] Built target base
Scanning dependencies of target mt
[ 19%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/CondVar.cpp.o
[ 20%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/Lock.cpp.o
[ 20%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/Mutex.cpp.o
[ 21%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/Thread.cpp.o
[ 21%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/XMT.cpp.o
[ 22%] Linking CXX static library ../../../../../lib/libmt.a
[ 22%] Built target mt
Scanning dependencies of target io
[ 22%] Building CXX object src/lib/io/CMakeFiles/io.dir/StreamBuffer.cpp.o
[ 23%] Building CXX object src/lib/io/CMakeFiles/io.dir/StreamFilter.cpp.o
[ 23%] Building CXX object src/lib/io/CMakeFiles/io.dir/XIO.cpp.o
[ 24%] Linking CXX static library ../../../../../lib/libio.a
[ 24%] Built target io
Scanning dependencies of target net
[ 25%] Building CXX object src/lib/net/CMakeFiles/net.dir/IDataSocket.cpp.o
[ 25%] Building CXX object src/lib/net/CMakeFiles/net.dir/NetworkAddress.cpp.o
[ 26%] Building CXX object src/lib/net/CMakeFiles/net.dir/SocketMultiplexer.cpp.o
[ 26%] Building CXX object src/lib/net/CMakeFiles/net.dir/TCPListenSocket.cpp.o
[ 27%] Building CXX object src/lib/net/CMakeFiles/net.dir/TCPSocket.cpp.o
[ 27%] Building CXX object src/lib/net/CMakeFiles/net.dir/TCPSocketFactory.cpp.o
[ 28%] Building CXX object src/lib/net/CMakeFiles/net.dir/XSocket.cpp.o
[ 28%] Linking CXX static library ../../../../../lib/libnet.a
[ 28%] Built target net
Scanning dependencies of target synergy
[ 29%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/App.cpp.o
[ 29%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/AppUtil.cpp.o
[ 30%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ArgParser.cpp.o
[ 30%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ArgsBase.cpp.o
[ 31%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/Chunk.cpp.o
[ 31%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ClientApp.cpp.o
[ 32%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ClientArgs.cpp.o
[ 32%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ClientTaskBarReceiver.cpp.o
[ 33%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/Clipboard.cpp.o
[ 34%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ClipboardChunk.cpp.o
[ 34%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/DaemonApp.cpp.o
[ 35%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/DpiHelper.cpp.o
[ 35%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/DragInformation.cpp.o
[ 36%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/DropHelper.cpp.o
[ 36%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/FileChunk.cpp.o
[ 37%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/IClipboard.cpp.o
[ 38%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/IKeyState.cpp.o
[ 38%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/IPrimaryScreen.cpp.o
[ 39%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/KeyMap.cpp.o
[ 39%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/KeyState.cpp.o
[ 40%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/PacketStreamFilter.cpp.o
[ 40%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/PlatformScreen.cpp.o
[ 41%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/PortableTaskBarReceiver.cpp.o
[ 41%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ProtocolUtil.cpp.o
[ 42%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/Screen.cpp.o
[ 43%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ServerApp.cpp.o
[ 43%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ServerArgs.cpp.o
[ 44%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ServerTaskBarReceiver.cpp.o
[ 44%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/StreamChunker.cpp.o
[ 45%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/SubscriptionManager.cpp.o
[ 45%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ToolApp.cpp.o
[ 46%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/ToolArgs.cpp.o
[ 47%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/XScreen.cpp.o
[ 47%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/XSynergy.cpp.o
[ 48%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/key_types.cpp.o
[ 48%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/protocol_types.cpp.o
[ 49%] Building CXX object src/lib/synergy/CMakeFiles/synergy.dir/unix/AppUtilUnix.cpp.o
[ 49%] Linking CXX static library ../../../../../lib/libsynergy.a
[ 49%] Built target synergy
Scanning dependencies of target server
[ 50%] Building CXX object src/lib/server/CMakeFiles/server.dir/BaseClientProxy.cpp.o
[ 50%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientListener.cpp.o
[ 51%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxy.cpp.o
[ 51%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxy1_0.cpp.o
[ 52%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxy1_1.cpp.o
[ 52%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxy1_2.cpp.o
[ 53%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxy1_3.cpp.o
[ 53%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxy1_4.cpp.o
[ 54%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxy1_5.cpp.o
[ 55%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxy1_6.cpp.o
[ 55%] Building CXX object src/lib/server/CMakeFiles/server.dir/ClientProxyUnknown.cpp.o
[ 56%] Building CXX object src/lib/server/CMakeFiles/server.dir/Config.cpp.o
[ 56%] Building CXX object src/lib/server/CMakeFiles/server.dir/InputFilter.cpp.o
[ 57%] Building CXX object src/lib/server/CMakeFiles/server.dir/PrimaryClient.cpp.o
[ 57%] Building CXX object src/lib/server/CMakeFiles/server.dir/Server.cpp.o
[ 58%] Linking CXX static library ../../../../../lib/libserver.a
[ 58%] Built target server
Scanning dependencies of target platform
[ 58%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsClipboard.cpp.o
[ 59%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsClipboardAnyBitmapConverter.cpp.o
[ 59%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsClipboardBMPConverter.cpp.o
[ 60%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsClipboardHTMLConverter.cpp.o
[ 61%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsClipboardTextConverter.cpp.o
[ 61%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsClipboardUCS2Converter.cpp.o
[ 62%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsClipboardUTF8Converter.cpp.o
[ 62%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsEventQueueBuffer.cpp.o
[ 63%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsKeyState.cpp.o
[ 63%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsScreen.cpp.o
[ 64%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsScreenSaver.cpp.o
[ 64%] Building CXX object src/lib/platform/CMakeFiles/platform.dir/XWindowsUtil.cpp.o
[ 65%] Linking CXX static library ../../../../../lib/libplatform.a
[ 65%] Built target platform
Scanning dependencies of target ipc
[ 66%] Building CXX object src/lib/ipc/CMakeFiles/ipc.dir/Ipc.cpp.o
[ 66%] Building CXX object src/lib/ipc/CMakeFiles/ipc.dir/IpcClient.cpp.o
[ 67%] Building CXX object src/lib/ipc/CMakeFiles/ipc.dir/IpcClientProxy.cpp.o
[ 67%] Building CXX object src/lib/ipc/CMakeFiles/ipc.dir/IpcLogOutputter.cpp.o
[ 68%] Building CXX object src/lib/ipc/CMakeFiles/ipc.dir/IpcMessage.cpp.o
[ 68%] Building CXX object src/lib/ipc/CMakeFiles/ipc.dir/IpcServer.cpp.o
[ 69%] Building CXX object src/lib/ipc/CMakeFiles/ipc.dir/IpcServerProxy.cpp.o
[ 69%] Linking CXX static library ../../../../../lib/libipc.a
[ 69%] Built target ipc
Scanning dependencies of target client
[ 70%] Building CXX object src/lib/client/CMakeFiles/client.dir/Client.cpp.o
[ 71%] Building CXX object src/lib/client/CMakeFiles/client.dir/ServerProxy.cpp.o
[ 71%] Linking CXX static library ../../../../../lib/libclient.a
[ 71%] Built target client
Scanning dependencies of target ns
[ 72%] Building CXX object src/lib/plugin/ns/CMakeFiles/ns.dir/SecureListenSocket.cpp.o
[ 73%] Building CXX object src/lib/plugin/ns/CMakeFiles/ns.dir/SecureSocket.cpp.o
[ 73%] Building CXX object src/lib/plugin/ns/CMakeFiles/ns.dir/ns.cpp.o
[ 74%] Linking CXX shared library ../../../../../../lib/libns.so
[ 74%] Built target ns
Scanning dependencies of target synergyc
[ 75%] Building CXX object src/cmd/synergyc/CMakeFiles/synergyc.dir/synergyc.cpp.o
[ 75%] Building CXX object src/cmd/synergyc/CMakeFiles/synergyc.dir/XWindowsClientTaskBarReceiver.cpp.o
[ 76%] Linking CXX executable ../../../../../bin/synergyc
[ 76%] Built target synergyc
Scanning dependencies of target synergys
[ 77%] Building CXX object src/cmd/synergys/CMakeFiles/synergys.dir/synergys.cpp.o
[ 77%] Building CXX object src/cmd/synergys/CMakeFiles/synergys.dir/XWindowsServerTaskBarReceiver.cpp.o
[ 78%] Linking CXX executable ../../../../../bin/synergys
[ 78%] Built target synergys
Scanning dependencies of target synergyd
[ 79%] Building CXX object src/cmd/synergyd/CMakeFiles/synergyd.dir/synergyd.cpp.o
[ 79%] Linking CXX executable ../../../../../bin/synergyd
[ 79%] Built target synergyd
Scanning dependencies of target micro
[ 80%] Building C object src/micro/CMakeFiles/micro.dir/uSynergy.c.o
[ 81%] Linking C static library ../../../../lib/libmicro.a
[ 81%] Built target micro
Scanning dependencies of target usynergy
[ 82%] Building C object src/cmd/usynergy/CMakeFiles/usynergy.dir/uSynergyUnix.c.o
[ 83%] Linking C executable ../../../../../bin/usynergy
[ 83%] Built target usynergy
Scanning dependencies of target syntool
[ 83%] Building CXX object src/cmd/syntool/CMakeFiles/syntool.dir/syntool.cpp.o
[ 84%] Linking CXX executable ../../../../../bin/syntool
[ 84%] Built target syntool
Scanning dependencies of target gmock
[ 84%] Building CXX object src/test/CMakeFiles/gmock.dir/__/__/ext/gmock-1.6.0/src/gmock-all.cc.o
[ 85%] Linking CXX static library ../../../../lib/libgmock.a
[ 85%] Built target gmock
Scanning dependencies of target gtest
[ 86%] Building CXX object src/test/CMakeFiles/gtest.dir/__/__/ext/gtest-1.6.0/src/gtest-all.cc.o
[ 86%] Linking CXX static library ../../../../lib/libgtest.a
[ 86%] Built target gtest
Scanning dependencies of target integtests
[ 87%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/Main.cpp.o
[ 87%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/arch/ArchInternetTests.cpp.o
[ 88%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/ipc/IpcTests.cpp.o
[ 88%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/net/NetworkTests.cpp.o
[ 89%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/platform/XWindowsClipboardTests.cpp.o
[ 89%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/platform/XWindowsKeyStateTests.cpp.o
[ 90%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/platform/XWindowsScreenSaverTests.cpp.o
[ 91%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/platform/XWindowsScreenTests.cpp.o
[ 91%] Building CXX object src/test/integtests/CMakeFiles/integtests.dir/__/global/TestEventQueue.cpp.o
[ 92%] Linking CXX executable ../../../../../bin/integtests
[ 92%] Built target integtests
Scanning dependencies of target unittests
[ 93%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/Main.cpp.o
[ 93%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/base/StringTests.cpp.o
[ 94%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/ipc/IpcLogOutputterTests.cpp.o
[ 94%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/ArgParserTests.cpp.o
[ 95%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/ClientArgsParsingTests.cpp.o
[ 95%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/ClipboardChunkTests.cpp.o
[ 96%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/ClipboardTests.cpp.o
[ 96%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/DeprecatedArgsParsingTests.cpp.o
[ 97%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/DpiHelperTests.cpp.o
[ 98%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/GenericArgsParsingTests.cpp.o
[ 98%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/KeyStateTests.cpp.o
[ 99%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/ServerArgsParsingTests.cpp.o
[ 99%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/synergy/SubscriptionTests.cpp.o
[100%] Building CXX object src/test/unittests/CMakeFiles/unittests.dir/__/global/TestEventQueue.cpp.o
[100%] Linking CXX executable ../../../../../bin/unittests
[100%] Built target unittests
Going back to: /mnt/data/build/synergy-v1.8.2-stable-36cd521-Source
Make GUI command: make -w
Entering dir: src/gui
make: Entering directory '/mnt/data/build/synergy-v1.8.2-stable-36cd521-Source/src/gui'
make: *** No targets specified and no makefile found.  Stop.
make: Leaving directory '/mnt/data/build/synergy-v1.8.2-stable-36cd521-Source/src/gui'
Going back to: /mnt/data/build/synergy-v1.8.2-stable-36cd521-Source
Error: make -w failed with error: 512
```

**Comment by faultylee** [comment]
Sorry, just realized this from `aur/synergy-git` [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=synergy-git)

I'm missing this 
```
  cd src/gui
  qmake
  make
```

Not sure if this applies to non Arch users, if it does, maybe the [compiling documentation](https://github.com/symless/synergy/wiki/Compiling#Unix_or_other_Linux) should be updated

**Comment by silvester747** [comment]
I noticed this happens if I download the source package. It seems to expect being in a git repo. If I clone the git repo, it works fine. So seems the source package is broken.

**Comment by nlyan** [comment]
The build method in the -git PKGBUILD isn't really the way we recommend building synergy (with the hm.py script).

I currently use Arch here at the Symless office so at some point I will probably request ownership of the packages and fix this. At the moment most of our Linux users are using deb and rpm distros though, so its not a priority. Proper repos for deb and rpm distributions are also on my personal todo list.

Closing for now.

**Comment by faultylee** [comment]
I understand about priority, though I feel that at least the documentation should be updated to reflect the actual steps needed to compile from source, it can be highlighted that it's not the recommended steps.

**Comment by silvester747** [comment]
Also on the guide referred to from the download page (https://github.com/symless/synergy/wiki/Compiling#Unix_Linux_Mac_OS_X), hm.sh is mentioned as the official way to build. So I would advice not closing this issue until the documentation is fixed.

**Comment by kpfleming** [comment]
Same issue here; building from a repository clone worked fine (using the wiki instructions), building from a source package failed to configure the gui subdirectory.

**Comment by anohren** [comment]
Having the same issue when trying to compile from github source zip.

**Comment by XinyuHou** [comment]
Wiki is updated to instruct users to download our source code from GitHub repo.

**Comment by jorhett** [comment]
So the account page with the download link still give you the source tarball if you browse from Linux.
   https://symless.com/account

There's nothing there to lead the person to the github README, thus leading everyone to go through failure before finding this bug

**Comment by freephile** [comment]
I'm a Synergy Pro paid user. I previously was happy with Synergy on Ubuntu 14.04. On Ubuntu 16.04, the download for Synergy (v1.8.8-stable-c30301e) is a tarball of the source, not a deb package that I can hand over to my system's package installer.  The INSTALL file, README and wiki compile instructions are out of sync with each other, outdated, or else do not give concrete steps for installing Synergy.  

I don't know what I'm supposed to do to install Synergy.

I try to use the hm.sh script, but I get errors like

Error: Could not get revision, git error: 128
Error: make -w failed with error: 512

./bin/usynergy 
Segmentation fault (core dumped)

What's more, when researching these errors, you find "bugs" like [5792](https://github.com/symless/synergy/issues/5792) which suggest starting off with a git clone; using forks; or changing to v2; or trying a bunch of different things to get configuration and compilation to work right.

I'm not trying to build the software as a developer, I have work to do and just want to use the software as a user between my Ubuntu 16.04 and Windows 8 machine.  Sadly, I can't do that.

I even registered on the Zendesk site where there is a thread about "[Getting Started](https://symless.zendesk.com/hc/en-us/articles/115004888708-Getting-started-with-Synergy)", updated just days ago by @nbolton Nick Bolton. But when I tried to submit a comment, it gives a 403 access denied!

**Comment by freephile** [comment]
Here's how I got Synergy installed and working on Ubuntu 16.04 (I already have the software prerequisites)
```
git clone https://github.com/symless/synergy.git
cd synergy
QT_SELECT=4 ./hm.sh conf -g1
```
It will spit out a qmake command that would be used to make the GUI.  I'm not sure if you need to execute this command, but I assume so.
```
cd src/gui/
qmake gui.pro -r QMAKE_VERSION_STAGE=stable QMAKE_VERSION_REVISION=ec56ac4
```
Launch the software
```
./bin/synergy
```
Make a valid configuration file by hand because the wizard doesn't detect the client that I want to connect to.
For example
```
cat ~/.synergy.conf
section: screens
  workstation:
  greg-laptop:
end
section: links
  workstation:
    left = greg-laptop
  greg-laptop:
    right = workstation
end
```
Now tell Synergy to use that config in the GUI; and start Synergy on the client, pointing to the IP address of the server.

**Comment by freephile** [comment]
I updated the [Compiling](https://github.com/symless/synergy/wiki/Compiling) page on the wiki

**Comment by gurcei** [comment]
I was running into the same issue when trying to build the downloadable tar.gz source from the main website. Rather than grab from git, I opted for this workaround:

1. I edited the "**ext/toolchain/commands1.py**" file
2. Inside the **getGitRevision(self)** method here, I simply returned a string mentioning the version-details of what I downloaded
3. Since I also wanted to build a package, I had to remedy the **getGitBranchName(self)** method in a similar fashion

E.g.:

```
def getGitRevision(self):
    return "v1.8.8-stable-c30301e-Source"

def getGitBranchName(self):
    return "unknown"
```

After doing so, I managed to do the following steps fine:

- QT_SELECT=4 ./hm.sh conf -g1
- ./hm.sh build
- ./hm.sh package rpm

This generates an rpm with a rather unwieldy name of "bin/**synergy-unknown-stable-v1.8.8-stable-c30301e-Source-Linux-i686.rpm**"

But despite it, it installed fine with "sudo rpm -i synergy-unknown-stable-v1.8.8-stable-c30301e-Source-Linux-i686.rpm"
